pytorch_lightning/trainer/trainer.py

```diff
@@ -835,6 +840,47 @@ def run_pretrain_routine(self, model: LightningModule):
         # CORE TRAINING LOOP
         self.train()
 
+    def validate(self, model: Optional[LightningModule] = None):
+        r"""
+
+        Separates from fit to make sure you never run on your test set until you want to.
+
+        Args:
+            model: The model to test.
+
+        Example::
+
+            # Option 1
+            # run validation after fitting
+            trainer = Trainer()
+            model = LightningModule()
+
+            trainer.fit()
+            trainer.validate()
+
+            # Option 2
+            # run validation from a loaded model
+            model = LightningModule.load_from_checkpoint('path/to/checkpoint.ckpt')
+            trainer = Trainer()
+            trainer.validate(model)
+        """
+        self.mode = TrainerMode.VALIDATING
+        if model is not None:
```

Comment:
This code and the code in test could be deduplicated with a helper function

Reply:
fyi #1195